File: LeetCode/1952. Three Divisors/1952-070821.java
import java.lang.Math.*;

/**
 * 1952. Three Divisors
 *
 * 07 August, 2021
 *
 * @author Hao Ren
 */
class Solution {
    public boolean isPrime(int x) {
        if (x < 2)
            return false;
        for (int i = 2; i * i <= x; i++) {
            if (x % i == 0)
                return false;
        }
        return true;
    }

    public boolean isThree(int n) {
        double root = Math.sqrt(n);
        if (root != (int) root)
            return false;
        return isPrime((int) root);
    }
}

File: LeetCode/1952. Three Divisors/1952-070821.py
"""
Hao Ren
07 August, 2021
1952. Three Divisors
"""

import math

class Solution(object):
    def is_prime(self, x):
        if x < 2:
            return False
        if x % 2 == 0 and x > 2:
            return False
        for i in range(3, int(math.sqrt(x)) + 1, 2):
            if x % i == 0:
                return False
        return True

    def isThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if (math.sqrt(n).is_integer() == False):
            return False
        return self.is_prime(math.sqrt(n))
